File: src/Solutions/Euler123.hs
module Solutions.Euler123 where
import Utils.List(everyOther)
import Utils.NumberTheory(unPrime, primes)
import Data.List
import Data.Maybe

isOK (n, p) = 2 * n * unPrime p > 10^10

euler123 :: IO String
euler123 = return $ show $ fst $ fromJust $ find isOK $ everyOther True $ zip [1..] primes

File: rust/src/solutions/euler123.rs
pub fn euler123() -> String {
    (1..)
        .zip(primal::Sieve::new(10000000).primes_from(0))
        .step_by(2)
        .find(|&(n, p)| 2 * n * p > 10_000_000_000)
        .unwrap()
        .0
        .to_string()
}
